File: suites/02_numerical_computation/comparisons/monte_carlo_pi.cpp
#include <cstdio>
static long long monte_carlo(long long samples) {
    long long inside = 0, seed = 123456789;
    for (long long i = 0; i < samples; ++i) {
        seed = ((seed * 1103515245LL) + 12345) % 2147483647;
        long long x = seed % 10000;
        seed = ((seed * 1103515245LL) + 12345) % 2147483647;
        long long y = seed % 10000;
        if ((x * x) + (y * y) <= 100000000) inside++;
    }
    return (inside * 4000000) / samples;
}
int main() { printf("%lld\n", monte_carlo(500000)); return 0; }

File: suites/02_numerical_computation/comparisons/monte_carlo_pi.py
def monte_carlo(samples: int) -> int:
    inside, seed = 0, 123456789
    for _ in range(samples):
        seed = ((seed * 1103515245) + 12345) % 2147483647
        x = seed % 10000
        seed = ((seed * 1103515245) + 12345) % 2147483647
        y = seed % 10000
        if (x * x) + (y * y) <= 100000000:
            inside += 1
    return (inside * 4000000) // samples

if __name__ == "__main__":
    print(monte_carlo(500000))
